WalletWasabi/Wallets/WalletManager.cs

```diff
@@ -159,7 +159,33 @@ public async Task<Wallet> StartWalletAsync(Wallet wallet)
 				await wallet.StartAsync(cancel).ConfigureAwait(false);
 				Logger.LogInfo($"Wallet '{wallet.WalletName}' started.");
 				cancel.ThrowIfCancellationRequested();
-
+				
+				// Continue wallet synchronization in the background for all keys skipped by TurboSync.
+				_ = Task.Run(
+					async () => 
+					{
+						while (!cancel.IsCancellationRequested)
```

Comment:
After you moved this inside `wallet.cs`, you should use `HandleFiltersLock` here, otherwise a filter can arrive during this with `IndexDownloader_NewFilterAsync` and could mess up the height, couldn't it?

Reply:
In wallet.cs you can store and manage the Task at StopAsync - so you do not forget it.